source/view/blocks.js

```diff
@@ -644,13 +645,68 @@ Blockly.Blocks[ 'controls_sensor_tracks' ] = {
     var thisBlock = this;
     this.setTooltip( function() {
       var content = {
-        text: "Checks our scanner for man-made objects and other anomalies."
+        text: "Checks our scanner for man-made objects and other anomalies immediately (one square) in front of the rover. Not available on Perry!"
       }
       return showTooltipInBlockly( thisBlock, content );
     } );
   }
 };
 
+Blockly.JavaScript[ 'controls_sensor_signal' ] = function( block ) {
+
+  var argument0 = Blockly.JavaScript.valueToCode(block, 'INPUT', Blockly.JavaScript.ORDER_ATOMIC) || '';
+
+  if ( roverSignalValue !== undefined ) {
+      return [ roverSignalValue + argument0, Blockly.JavaScript.ORDER_ATOMIC ];
+  } else {
+      return [ ' 0 ' + argument0, Blockly.JavaScript.ORDER_ATOMIC ];
+  }
+ 
+  
+};
+
+Blockly.Blocks[ 'controls_sensor_signal' ] = {
+  init: function() {
+    this.setColour( 30 );
+    this.appendValueInput('INPUT')
+        .appendField('Base Signal °')
+        .setCheck(['OperatorAddSubtract','OperatorMultiplyDivide','LeftParenthesis','RightParenthesis','Conditional']);
+    this.setOutput(true, null);
+
+    var thisBlock = this;
+    this.setTooltip( function() {
+      var content = {
+        text: "Checks our scanner for base location in a 360° arc that starts on the X-axis. Not yet available on Perry!"
```

Comment:
Also, I might specify that it returns a value between 0 and 360 (or -180 and 180, or whatever).